### ISSUE TITLE
Issue with wercker/default (HTTP 404)

### ISSUE THREAD

**Issue by herveDarritchon** [ArjenSchwarz/wercker-step-hugo-build#8]
This container seems to don't exist anymore ?

when I run wercker build with this wercker.yml

```
    box: wercker/default
    build:
      steps:
        - arjen/hugo-build:
            version: 0.14
            theme: hd-theme
            flags: --disableSitemap=true
```

I get an HTTP 404 error :
--> Executing pipeline
--> Running step: setup environment
ERROR Error during check access
ERROR --> Step failed: setup environment
GET https://registry.hub.docker.com/v1/repositories/wercker/default/images returned 404
ERROR GET https://registry.hub.docker.com/v1/repositories/wercker/default/images returned 404

**Comment by ArjenSchwarz** [comment]
Hello @herveDarritchon, 
Thanks for letting me know about this. 
I believe this might be caused by a change on Wercker's side where they have changed their defaults to use Docker instead of the boxes they used before. I've been planning on updating the step and documentation to ensure it can properly deal with that, but a holiday and then getting sick prevented me from doing so. I'll do my best to look into this and your other report to try and fix it tonight (which is about 12 hours from now), and will let you know once I've done so. 
You might be able to use their old way of doing things until then, but I suspect it will be more useful longterm to use their new tools as that will be more future proof.

**Comment by herveDarritchon** [comment]
Don't worry @ArjenSchwarz 

I have a workaround, I use the python/weezy container instead. 

But what is weird, it's that even in the wercker official documentation, they say you can use the wercker/default docker container (http://devcenter.wercker.com/docs/wercker-yml/wercker-classic.html)

But it's not working ...

don't know why

**Comment by ArjenSchwarz** [comment]
Hi @herveDarritchon I've looked into this a bit, and I do believe my initial idea was correct that this is indeed because they changed to using Docker. Let me explain it a bit so it's more clear.

Initially Wercker used their own Docker like system with "boxes", this is what is now referred to as wercker classic and is what the documentation you mentioned explains. However, they've been testing with using Docker itself for a while now and not long ago they made that the default. You can see this when you create a new Wercker application where it will have the "Docker enabled" checkbox in step 5 enabled. 

Toggling between these two states for a repo without a wercker file (or anything really) present will show the example default boxes, which is indeed `wercker/default` for classic, but a `debian` box when using Docker. Therefore I presume that they didn't create a `wercker/default` Docker container (which would only have been a copy of the Debian one) which then doesn't match my documentation anymore.

Let me know if this isn't clear, and I'll try to explain it some more. I'll leave this issue open until I've updated the documentation.

**Comment by herveDarritchon** [comment]
it's totally clear :)
I agree.

In the case of the Debian docker image, it misses some command like wget so you can't use it as it is. You need to enhance it at least with wget installed in it.

**Comment by ArjenSchwarz** [comment]
Hey @herveDarritchon, I've updated the plugin to work correctly with the Docker setup. By that I mean that the documentation has been updated, but also that it is now able to download Hugo even if no curl is available (which it now uses instead of wget as it's more likely to be installed on Docker containers).

I will create a pull request for updating the tutorial in the Hugo documentation as well, but that is separate to this so I will now close this issue.
